python/dlr/counter/counter_mgr.py

```diff
@@ -0,0 +1,144 @@
+import platform
+import atexit
+import json
+import hashlib
+import os
+import logging
+
+from .publisher import MsgPublisher
+from .system import Factory
+from . import config
+from .model_exec_counter import ModelExecCounter
+from .model_metric import ModelMetric
+
+
+def call_home(func):
+    def wrapped_call_home(*args, **kwargs):
+        call_counter = CallCounterMgr.get_instance()
+        if call_counter:
+            if func.__name__ == "init_call_home":
+                func(*args, **kwargs)
+                call_counter.runtime_loaded()
+            elif func.__name__ == "__init__":
+                func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_loaded(obj.get_model_name())
+            else:
+                res = func(*args, **kwargs)
+                obj = args[0]
+                call_counter.model_run(obj.get_model_name())
+                return res
+
+    return wrapped_call_home
+
+
+class CallCounterMgr(object):
+    RUNTIME_LOAD = 1
+    MODEL_LOAD = 2
+    MODEL_RUN = 3
+    _instance = None
+
+    @staticmethod
+    def get_instance():
+        """return single instance of class"""
+        if CallCounterMgr._instance is None:
+            if CallCounterMgr.is_feature_enabled():
+                print(config.CALL_HOME_USR_NOTIFICATION)
+                CallCounterMgr._instance = CallCounterMgr()
+                atexit.register(CallCounterMgr._instance.stop)
+            else:
+                logging.warning("call home feature disabled")
+        return CallCounterMgr._instance
+
+    def __init__(self):
+        try:
+            self.msg_publisher = MsgPublisher.get_instance()
+            machine_typ = platform.machine()
+            os_name = platform.system()
+            os_supt = "{0}_{1}".format(os_name, machine_typ)
+            self.system = Factory.get_system(os_supt)
+            self.model_metric = ModelMetric.get_instance(self.system.get_device_uuid())
+        except Exception as e:
+            logging.exception("while in counter mgr init", exc_info=True)
+
+    @staticmethod
+    def is_feature_enabled():
+        """check feature disabled config. ccm_config.json file present in root folder"""
+        try:
+            if os.path.isfile(config.CALL_HOME_USER_CONFIG_FILE):
+                with open(config.CALL_HOME_USER_CONFIG_FILE, "r") as ccm_json_file:
+                    data = json.load(ccm_json_file)
+                    if 'false' == str(data['ccm']).lower():
+                        return False
+                    else:
+                        return True
+            else:
+                return True
+        except Exception as e:
+            logging.exception("while in reading ccm config file")
+        return True
```

Comment:
I want this to be a static hard-coded variable, DEFAULT_ENABLE_FEATURE = True.

Reason being this value is yet decided, and I don't want the hardcoded True as return value

Reply:
Added static variable to hold default configuration of feature True.